### ISSUE TITLE
id_accuracy=0

### ISSUE THREAD

**Issue by jjn037** [ShuangLI59/person_search#69]
I1122 20:34:16.939185  8871 sgd_solver.cpp:105] Iteration 34120, lr = 0.001
I1122 20:34:25.444725  8871 solver.cpp:219] Iteration 34140 (2.35148 iter/s, 8.50528s/20 iters), loss = 5.27471
I1122 20:34:25.444803  8871 solver.cpp:238]     Train net output #0: det_accuracy = 0.953125
I1122 20:34:25.444819  8871 solver.cpp:238]     Train net output #1: det_loss = 0.116758 (* 1 = 0.116758 loss)
I1122 20:34:25.444828  8871 solver.cpp:238]     Train net output #2: id_accuracy = 0
I1122 20:34:25.444836  8871 solver.cpp:238]     Train net output #3: id_loss = 7.04522 (* 1 = 7.04522 loss)
I1122 20:34:25.444845  8871 solver.cpp:238]     Train net output #4: loss_bbox = 0.379949 (* 1 = 0.379949 loss)
I1122 20:34:25.444854  8871 solver.cpp:238]     Train net output #5: rpn_cls_loss = 0.0538092 (* 1 = 0.0538092 loss)
I1122 20:34:25.444862  8871 solver.cpp:238]     Train net output #6: rpn_loss_bbox = 0.0670886 (* 1 = 0.0670886 loss)
I1122 20:34:25.444875  8871 sgd_solver.cpp:105] Iteration 34140, lr = 0.001
I1122 20:34:33.161725  8871 solver.cpp:219] Iteration 34160 (2.59181 iter/s, 7.71661s/20 iters), loss = 5.08991
I1122 20:34:33.161808  8871 solver.cpp:238]     Train net output #0: det_accuracy = 0.929688
I1122 20:34:33.161828  8871 solver.cpp:238]     Train net output #1: det_loss = 0.10782 (* 1 = 0.10782 loss)
I1122 20:34:33.161839  8871 solver.cpp:238]     Train net output #2: id_accuracy = 0
I1122 20:34:33.161851  8871 solver.cpp:238]     Train net output #3: id_loss = 6.10264 (* 1 = 6.10264 loss)
I1122 20:34:33.161862  8871 solver.cpp:238]     Train net output #4: loss_bbox = 0.17604 (* 1 = 0.17604 loss)
I1122 20:34:33.161873  8871 solver.cpp:238]     Train net output #5: rpn_cls_loss = 0.127531 (* 1 = 0.127531 loss)
I1122 20:34:33.161885  8871 solver.cpp:238]     Train net output #6: rpn_loss_bbox = 0.0870427 (* 1 = 0.0870427 loss)
I1122 20:34:33.161900  8871 sgd_solver.cpp:105] Iteration 34160, lr = 0.001
I1122 20:34:40.963131  8871 solver.cpp:219] Iteration 34180 (2.56377 iter/s, 7.801s/20 iters), loss = 5.09691
I1122 20:34:40.963213  8871 solver.cpp:238]     Train net output #0: det_accuracy = 0.929688
I1122 20:34:40.963233  8871 solver.cpp:238]     Train net output #1: det_loss = 0.163159 (* 1 = 0.163159 loss)
I1122 20:34:40.963245  8871 solver.cpp:238]     Train net output #2: id_accuracy = 0.3
I1122 20:34:40.963258  8871 solver.cpp:238]     Train net output #3: id_loss = 3.88689 (* 1 = 3.88689 loss)
I1122 20:34:40.963268  8871 solver.cpp:238]     Train net output #4: loss_bbox = 0.222411 (* 1 = 0.222411 loss)
I1122 20:34:40.963279  8871 solver.cpp:238]     Train net output #5: rpn_cls_loss = 0.202602 (* 1 = 0.202602 loss)
I1122 20:34:40.963289  8871 solver.cpp:238]     Train net output #6: rpn_loss_bbox = 0.993391 (* 1 = 0.993391 loss)
I1122 20:34:40.963304  8871 sgd_solver.cpp:105] Iteration 34180, lr = 0.001
I1122 20:34:49.098305  8871 solver.cpp:219] Iteration 34200 (2.45858 iter/s, 8.13476s/20 iters), loss = 5.19017
I1122 20:34:49.098373  8871 solver.cpp:238]     Train net output #0: det_accuracy = 0.984375
I1122 20:34:49.098388  8871 solver.cpp:238]     Train net output #1: det_loss = 0.0294993 (* 1 = 0.0294993 loss)
I1122 20:34:49.098397  8871 solver.cpp:238]     Train net output #2: id_accuracy = 0.2
I1122 20:34:49.098404  8871 solver.cpp:238]     Train net output #3: id_loss = 4.19861 (* 1 = 4.19861 loss)
I1122 20:34:49.098413  8871 solver.cpp:238]     Train net output #4: loss_bbox = 0.0517187 (* 1 = 0.0517187 loss)
I1122 20:34:49.098422  8871 solver.cpp:238]     Train net output #5: rpn_cls_loss = 0.0103175 (* 1 = 0.0103175 loss)
I1122 20:34:49.098429  8871 solver.cpp:238]     Train net output #6: rpn_loss_bbox = 0.00192679 (* 1 = 0.00192679 loss)
I1122 20:34:49.098440  8871 sgd_solver.cpp:105] Iteration 34200, lr = 0.001
I1122 20:34:57.153692  8871 solver.cpp:219] Iteration 34220 (2.48293 iter/s, 8.05499s/20 iters), loss = 5.04621
I1122 20:34:57.153774  8871 solver.cpp:238]     Train net output #0: det_accuracy = 0.976562
I1122 20:34:57.153792  8871 solver.cpp:238]     Train net output #1: det_loss = 0.0611082 (* 1 = 0.0611082 loss)
I1122 20:34:57.153805  8871 solver.cpp:238]     Train net output #2: id_accuracy = -nan
I1122 20:34:57.153815  8871 solver.cpp:238]     Train net output #3: id_loss = 0 (* 1 = 0 loss)
I1122 20:34:57.153827  8871 solver.cpp:238]     Train net output #4: loss_bbox = 0.120212 (* 1 = 0.120212 loss)
I1122 20:34:57.153837  8871 solver.cpp:238]     Train net output #5: rpn_cls_loss = 0.0470721 (* 1 = 0.0470721 loss)
I1122 20:34:57.153849  8871 solver.cpp:238]     Train net output #6: rpn_loss_bbox = 0.00547891 (* 1 = 0.00547891 loss)
I1122 20:34:57.153864  8871 sgd_solver.cpp:105] Iteration 34220, lr = 0.001
^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[BI1122 20:35:04.813046  8871 solver.cpp:219] Iteration 34240 (2.61132 iter/s, 7.65896s/20 iters), loss = 4.89673
I1122 20:35:04.813129  8871 solver.cpp:238]     Train net output #0: det_accuracy = 1
I1122 20:35:04.813149  8871 solver.cpp:238]     Train net output #1: det_loss = 0.022339 (* 1 = 0.022339 loss)
I1122 20:35:04.813160  8871 solver.cpp:238]     Train net output #2: id_accuracy = -nan
I1122 20:35:04.813174  8871 solver.cpp:238]     Train net output #3: id_loss = 0 (* 1 = 0 loss)
I1122 20:35:04.813184  8871 solver.cpp:238]     Train net output #4: loss_bbox = 0.126798 (* 1 = 0.126798 loss)
I1122 20:35:04.813195  8871 solver.cpp:238]     Train net output #5: rpn_cls_loss = 0.02272 (* 1 = 0.02272 loss)
I1122 20:35:04.813206  8871 solver.cpp:238]     Train net output #6: rpn_loss_bbox = 0.0226682 (* 1 = 0.0226682 loss)
I1122 20:35:04.813221  8871 sgd_solver.cpp:105] Iteration 34240, lr = 0.001
I1122 20:35:13.222926  8871 solver.cpp:219] Iteration 34260 (2.37827 iter/s, 8.40946s/20 iters), loss = 5.04766
I1122 20:35:13.222995  8871 solver.cpp:238]     Train net output #0: det_accuracy = 0.875
I1122 20:35:13.223009  8871 solver.cpp:238]     Train net output #1: det_loss = 0.255105 (* 1 = 0.255105 loss)
I1122 20:35:13.223017  8871 solver.cpp:238]     Train net output #2: id_accuracy = 0
I1122 20:35:13.223027  8871 solver.cpp:238]     Train net output #3: id_loss = 4.59072 (* 1 = 4.59072 loss)
I1122 20:35:13.223036  8871 solver.cpp:238]     Train net output #4: loss_bbox = 0.393796 (* 1 = 0.393796 loss)
I1122 20:35:13.223043  8871 solver.cpp:238]     Train net output #5: rpn_cls_loss = 0.0156542 (* 1 = 0.0156542 loss)
I1122 20:35:13.223052  8871 solver.cpp:238]     Train net output #6: rpn_loss_bbox = 0.00982659 (* 1 = 0.00982659 loss)

I added your labled_matching_layer.* and labled_matching_layer.* into https://github.com/soeaver/py-RFCN-priv/tree/master/caffe-priv, but the others like lib/ tools/ experiment/ use yours, and this code doesn't have mpi and my train.prototxt have not 
mem_param {
  optimize_train: true
  optimize_test: true
}

I have successed to make caffe_priv after add the two layer, but when I try to train my model, the id_accuracy seems to get some problems

**Comment by jjn037** [comment]
And I wonder that if 
mem_param {
optimize_train: true
optimize_test: true
} 
is important for oim part, and if I have added another files for oim part

**Comment by Cysu** [comment]
The mem_param is for memory optimization, which reduces the memory consumption by half. As long as you can train the net with your caffe, you can ignore them.

I am not sure what the data you are using, but you may tweak the scale (reciprocal of the temperature) befor the softmax layer. It might help ease the training.
